### PR TITLE
create endpoint for get /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by Kayphaz007**

### CODE REVIEW

File: models/articles.models.js [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
+      }
+    });
+}
+
+function selectAllCommentsByArticleId(article_id) {
+    return db.query('SELECT * FROM comments WHERE article_id = $1', [article_id]).then(({rows})=>{
+        if (rows.length != 0){
+            return rows
+        } else {
+            return Promise.reject({ status: 404, msg: "Not Found"})
```
Comment (robert112233) [inline models/articles.models.js]: Your Promise.reject will error for any valid articles with no comments, this situation is a 200 as nothing has gone wrong, you may want to invoke a check exists before you reject the promise

File: __tests__/app.test.js [hunk 2]
```diff
@@ -121,4 +121,30 @@ describe("200: /api/articles", () => {
...
+  });
+  test("should return error with msg Not Found for request not in database", () => {
+    return request(app)
+      .get("/api/articles/99999999/comments")
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Not Found");
+      });
+  });
```
Comment (robert112233) [inline __tests__/app.test.js:167]: Missing tests:

400 - Invalid request /api/articles/notanid/comments
200 - Valid article with no comments (returns [])

**Review by robert112233 — CHANGES_REQUESTED** [review]
Great start Micheal, implement the missing tests and submit another PR for review 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -138,13 +138,22 @@ describe("200: /api/articles", () => {
...
+  test("should return 200 with empty array for request with no comments", () => {
     return request(app)
-      .get("/api/articles/99999999/comments")
-      .expect(404)
+      .get("/api/articles/37/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toEqual([]);
+      });
```
Comment (robert112233) [inline __tests__/app.test.js:148]: This test is great and what I asked for, but you replaced an older test that we still need! 

 test("should return error with msg Not Found for request not in database", () => {
    return request(app)
      .get("/api/articles/99999999/comments")
      .expect(404)
      } 
      
You need to implement both of these at the same time, differentiating between existing articles with no comments, and non existent articles, _both_ of these situations return empty rows [] so you need to create a function that checks the article exists first :)

**Review by robert112233 — CHANGES_REQUESTED** [review]
You've misinterpreted my feedback, you will need a checkExists function in your code, https://notes.northcoders.com/courses/js-back-end/advanced-error-handling#sending-the-correct-response

**Review by roisinoreilly — APPROVED** [review]
nicely refactored :) great work, PR approved

**Pull request merged**